### PR TITLE
Amy's Debugging Challenge

### PR DESCRIPTION
**Pull request by amymariewall**

### CODE REVIEW

File: app/models/student.rb [hunk 1]
```diff
@@ -41,28 +41,28 @@ def overall_grade_adjusted
...
     self.grades.each do |g|
-      arr = g.adjusted_score
+      arr << g.adjusted_score
     end
     return arr
   end
 
   def grade_for_assignment(assignment_object)
-    submission = self.submissions.find_by_assignment_id(assignment_object.id)
+    @submission = self.submissions.find_by_assignment_id(assignment_object.id)
```
Comment (sumeetjain) [inline app/models/student.rb]: Was this change necessary? What happens if you remove the `@` symbols?

**Comment by amymariewall** [comment]
I think it is necessary - when I remove it I get this error: 

"NameError at /students/1
undefined local variable or method `submission' for #<Student id: 1, name: "Diana Mkrtchyan"> Did you mean? submissions submissions=
file: attribute_methods.rb location: method_missing line: 433"

**Comment by sumeetjain** [comment]
Did you remove all of the ones you added? If so, please commit that so I can see it.

**Comment by amymariewall** [comment]
Yes, I removed them all. I found an alternative fix that does _not_ involve making submission an instance variable. Now I think it is very odd that that 'fix' of adding the @ in front of 'submission' ever worked at all.

**Comment by sumeetjain** [comment]
That looks good! :+1:
